### ISSUE TITLE
.NET 5: can't build F# projects

### ISSUE THREAD

**Issue by rolfbjarne** [dotnet/sdk#12954]
Test files:

* AssemblyInfo.fs

```fsharp
namespace fsharplibrary
open System.Reflection

[<assembly: AssemblyVersion("1.0.0.0")>]

()
```

* fsharlibrary.fsproj

```xml
<?xml version="1.0" encoding="utf-8"?>
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netcoreapp5.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <Compile Include="AssemblyInfo.fs" />
  </ItemGroup>
</Project>
```

* Build:

```shell
$ dotnet build /nologo
  Determining projects to restore...
  All projects are up-to-date for restore.
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
/usr/local/share/dotnet/sdk/5.0.100-rc.1.20411.10/Microsoft.Common.CurrentVersion.targets(4207,5): error : Expected file "obj/Debug/netcoreapp5.0/ref/fsharplibrary.dll" does not exist. [/Users/rolf/test/dotnet/fsharplibrary/fsharplibrary.fsproj]

Build FAILED.

/usr/local/share/dotnet/sdk/5.0.100-rc.1.20411.10/Microsoft.Common.CurrentVersion.targets(4207,5): error : Expected file "obj/Debug/netcoreapp5.0/ref/fsharplibrary.dll" does not exist. [/Users/rolf/test/dotnet/fsharplibrary/fsharplibrary.fsproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.75
```

Zipped test case: 
[fsharplibrary-d885c90.zip](https://github.com/dotnet/sdk/files/5090095/fsharplibrary-d885c90.zip)

.NET version (on macOS 10.15.6): 

```shell
$ dotnet --version
5.0.100-rc.1.20411.10
```

It looks like this regressed in https://github.com/dotnet/sdk/pull/12720, if I comment out the line that sets `ProduceReferenceAssembly` to true then the build works.

**Comment by dsplaisted** [comment]
The workaround should be to set the `ProduceReferenceAssembly` property to false.

**Comment by rolfbjarne** [comment]
> The workaround should be to set the `ProduceReferenceAssembly` property to false.

Yes, that works, and that's what I did.

**Comment by sfoslund** [comment]
This was fixed by #13085
